go/vt/discovery/tablet_stats_cache.go

```diff
@@ -142,10 +146,14 @@ func (tc *TabletStatsCache) getOrCreateEntry(target *querypb.Target) *tabletStat
 	return e
 }
 
+func (tc *TabletStatsCache) getRegionByCell(cell string) string {
+	return topo.GetRegionByCell(context.Background(), tc.ts, cell)
+}
+
 // StatsUpdate is part of the HealthCheckStatsListener interface.
 func (tc *TabletStatsCache) StatsUpdate(ts *TabletStats) {
-	if ts.Target.TabletType != topodatapb.TabletType_MASTER && ts.Tablet.Alias.Cell != tc.cell {
-		// this is for a non-master tablet in a different cell, drop it
+	if ts.Target.TabletType != topodatapb.TabletType_MASTER && ts.Tablet.Alias.Cell != tc.cell && tc.getRegionByCell(ts.Tablet.Alias.Cell) != tc.getRegionByCell(tc.cell) {
```

Comment:
Shouldn't you check for blank region here? Otherwise, if region is not specified, it will end up being false always, and end up including all cells in the stats.

Reply:
actually not, as the `getRegionByCell` won't ever return empty region, it returns the `cell` value in this case. as agreed for backward compatibility

Reply:
of course :)